var/spack/repos/builtin/packages/libzmq/package.py

```diff
@@ -5,120 +5,131 @@
 
 import sys
 
+from spack.build_systems import autotools
 from spack.package import *
 
 
 class Libzmq(AutotoolsPackage):
     """The ZMQ networking/concurrency library and core API"""
 
     homepage = "https://zguide.zeromq.org/"
-    url = "https://github.com/zeromq/libzmq/releases/download/v4.3.5/zeromq-4.3.5.tar.gz"
     git = "https://github.com/zeromq/libzmq.git"
 
+    def url_for_version(self, ver):
+        if ver <= Version("4.1.4"):
+            return f"http://download.zeromq.org/zeromq-{ver}.tar.gz"
+        return f"https://github.com/zeromq/libzmq/releases/download/v{ver}/zeromq-{ver}.tar.gz"
+
     maintainers("dennisklein")
 
+    conflicts("%gcc@8:", when="@:4.2.2")
```

Comment:
versions usually go first, can you leave that?

Reply:
moved it down again just above the `depends_on`s